restapi.go

```diff
@@ -699,13 +699,8 @@ func (s *Session) GuildMemberDeleteWithReason(guildID, userID, reason string) (e
 // GuildMemberEdit edits the roles of a member.
 // guildID  : The ID of a Guild.
 // userID   : The ID of a User.
-// roles    : A list of role ID's to set on the member.
-func (s *Session) GuildMemberEdit(guildID, userID string, roles []string) (err error) {
-
-	data := struct {
-		Roles []string `json:"roles"`
-	}{roles}
-
+// data     : Struct of GuildMemberEditData
+func (s *Session) GuildMemberEdit(guildID, userID string, data *GuildMemberEditData) (err error) {
```

Comment:
I think still would be better to name it `GuildMemberEditComplex`. To avoid breaking current code.